tmt/trying.py

```diff
@@ -388,6 +399,28 @@ def action_quit(self, plan: Plan) -> None:
         run_id = click.style(plan.my_run.workdir, fg="magenta")
         self.print(f"Run {run_id} successfully finished. Bye for now!")
 
+    def handle_options(self, plan: Plan) -> None:
+        """Choose requested cli option"""
+
+        for o in self.cli_options:
+            if self.opt(o):
+                getattr(self, f"handle_{o}")(plan)
```

Comment:
Maybe a decorator? These handlers take a single parameter, type-wise it should be trivial to set up, and it could collect handlers for options.

On the other hand, `getattr()` & `handle_$option` is not that uncommon approach, see various examples of visitor design pattern in (Python) libraries, namely libraries processing some nodes and trees - XML, or markup languages in general, like docutils and parsign and rendering of ReST in https://github.com/teemtee/tmt/blob/main/tmt/utils/__init__.py#L7329. I'm not saying it's the best possible option, but it's not uncommon and if documented, it would be familiar to many developers.

I wouldn't block on this change, we can expect development in this area when new options are added, and decide to improve if needed.

Reply:
Thanks @happz, I agree.